examples/hyperparamopt/hpo_example.py

```diff
@@ -0,0 +1,137 @@
+##############################
+
+# Define a space for hpo to use
+# The space needs to define
+# 1. Name of the variables
+# 2. Default samplers for the variables (use scipy.stats objects)
+# 3. lo and hi ranges for the variables (will use -inf, inf if not specified)
+space = {'x1':{'dist': st.uniform(x1lo, x1hi-x1lo), 'lo':x1lo, 'hi':x1hi},
+         'x2':{'dist': st.uniform(x2lo, x2hi-x2lo), 'lo':x2lo, 'hi':x2hi}}
+
+# The trials object is just a list that stores the samples generated and the
+# corresponding function values at those sample points.
+trials = []
+
+# Maximum number of samples that will be generated.
+# This is the maximum number of function evaluations that will be performed.
+n_hpo_samples = 100
+
+# Call the fmin function that does the optimization.
+# The function to be optimized should take in a dictionary. You will probably
+# need to wrap your function to do this (see branin() and branin_wrapper()).
+# You can pass in a non-empty trials object as well e.g. from a previous
+# fmin run. We just append to the trials object and will use existing data
+# in our optimization.
+print("Starting optimization through hpo")
+best = hpo.fmin(loss_fn=branin_wrapper, space=space,
+                max_evals=n_hpo_samples, trials=trials)
+
+# Print out the best value obtained through HPO
+print("Best obtained through HPO (", n_hpo_samples, " samples) = ",
+       best['x1'], best['x2'], "; min value = ", best['loss'])
+
+#####################################
+# Optimization through grid search
+#####################################
+
+# Divide the space into a uniform grid (meshgrid)
+n = 200
+x1 = np.linspace(x1lo, x1hi, n)
+x2 = np.linspace(x2lo, x2hi, n)
+x1_grid, x2_grid = np.meshgrid(x1, x2)
+
+# Calculate the function values along the grid
+print("Starting optimization through grid search")
+z = branin(x1_grid, x2_grid)
+
+# Print out the best value obtained through grid search
+print("Best obtained through grid search (", n*n, " samples) = ",
+       x1_grid.flatten()[z.argmin()], x2_grid.flatten()[z.argmin()],
+       "; min value = ", z.min())
+
+########
+# Plots
+########
+
+# Convert trials object data into numpy arrays
+x1 = np.array([tr['x1'] for tr in trials])
+x2 = np.array([tr['x2'] for tr in trials])
+y = np.array([tr['loss'] for tr in trials])
+
+# Plot the function contour using the grid search data
+h = (z.max()-z.min())/25
+plt.contour(x1_grid, x2_grid, z, levels=np.linspace(z.min()-h, z.max(), 26))
+
+# Mark the points that were sampled through HPO
+plt.scatter(x1, x2, s=10, color='r', label='HPO Samples')
+
+# Mark the best points obtained through both methods
+plt.scatter(best['x1'], best['x2'], s=30, color='b', label='Best HPO')
+plt.scatter(x1_grid.flatten()[z.argmin()], x2_grid.flatten()[z.argmin()],
+            s=30, color='g', label='Best grid search')
+
+# Labels
+plt.xlabel('x1')
+plt.ylabel('x2')
+plt.title('Hyperparamter optimization using HPO (Branin function)')
+plt.legend()
+plt.show()
```

Comment:
Probably. The code should print the points. Are they very close to each other? 
Or try running again. It should randomly generate another set of points.